code/discovery.py

```diff
@@ -159,7 +159,10 @@ def getCurrentImageVersion(client):
     cudaCoreVersion = ''
     
     for container in client.containers.list():
-        img, tag = container.image.attrs['RepoTags'][0].split(':')
+        repotags = container.image.attrs.get('RepoTags')
+        if not repotags:
+            continue
+        img, tag = repotags[0].split(':')
```

Comment:
shouldn't there be a `break` below? If we find the peripheral-manager-gpu image, then we don't need to keep looping through the other containers right?

Reply:
This loop is used for getting both the `peripheralVersion` and `cudaCoreVersion` variables (from two different containers).
This is why it's a `continue` instead of a `break`.

Reply:
that's not it. 

Let's put it this way:
```
if not repotags:
 continue
else:
 img, tag = ....
 # you already have the img and tag, so no need to look further
 break
```